sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -2427,7 +2428,8 @@ ProgramManager::link(const device_image_plain &DeviceImage,
   };
   ur_result_t Error = doLink();
   if (Error == UR_RESULT_ERROR_OUT_OF_RESOURCES ||
-      Error == UR_RESULT_ERROR_OUT_OF_HOST_MEMORY) {
+      Error == UR_RESULT_ERROR_OUT_OF_HOST_MEMORY ||
```

Comment:
That is 3 places in code to be aware of, I feel like this `if` should be outlined into a helper function like `shouldWeClearInMemoryCacheAndRetry` so that next time we find a one more error code like this we only have to make changes in a single place